hyperspy/io_plugins/blockfile.py

```diff
@@ -0,0 +1,273 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2015 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from datetime import datetime, timedelta
+from dateutil import tz
+from traits.api import Undefined
+import numpy as np
+import logging
+import warnings
+
+from hyperspy.misc.array_tools import sarray2dict, dict2sarray
+
+_logger = logging.getLogger(__name__)
+
+
+# Plugin characteristics
+# ----------------------
+format_name = 'Blockfile'
+description = 'Read/write support for ASTAR blockfiles'
+full_support = False
+# Recognised file extension
+file_extensions = ['blo', 'BLO']
+default_extension = 0
+
+# Writing capabilities:
+# writes = False
+writes = [(2, 2)]
```

Comment:
Not as a different type, but I guess we can upconvert (1,Z,Z) and (X,Z,Z) to simply having NX and/or NY as 1.